modin/pandas/groupby.py

```diff
@@ -914,6 +960,9 @@ def _wrap_aggregation(
         else:
             groupby_qc = self._query_compiler
 
+        if all(not is_numeric_dtype(dtype) for dtype in groupby_qc.dtypes):
```

Comment:
shouldn't this be `any(...)`?..

Reply:
No, because we should follow logic that `numeric_only=False` in case ALL columns aren't numeric, but not ANY. Logic is aligned with pandas.

Reply:
If I'm reading pandas docs correctly, `numeric_only=None` (they say nothing about `False`) means "use all columns", and `True` means "use only numeric".

If this is true, I still think it should be `any()`, because a single numeric column in your case would cause all others to be dropped.

Reply:
`numeric_only=None` means "resolve bool value internally". pandas resolves `numeric_only=None` internally here https://github.com/pandas-dev/pandas/blob/835c78eb627dc30e800f25271be5fb9a8c0d4070/pandas/core/groupby/groupby.py#L2091-L2094. This is the logic of resolver: https://github.com/pandas-dev/pandas/blob/835c78eb627dc30e800f25271be5fb9a8c0d4070/pandas/core/groupby/groupby.py#L1195-L1232 

For example:

If dataframe with at least ONE numeric column is used, all others to be dropped in the result:
```python
import pandas as pd

data = {
    "A": ["0.0", "1.0", "2.0", "3.0", "4.0"],
    "B": ["0.0", "1.0", "0.0", "1.0", "0.0"],
    "C": ["foo1", "foo2", "foo3", "foo4", "foo5"],
    "D": [0, 1, 2, 3, 4],
}
pandas_df = pd.DataFrame(data)
print(f"pandas dtypes:\n {pandas_df.dtypes}")

pandas_res = pandas_df.groupby("B").sum()
print(pandas_res)
```
If dataframe with at least ONE numeric column (`D` in this case) is used, all others to be dropped in the result:
```bash
pandas dtypes:
 A    object
B    object
C    object
D     int64
dtype: object
     D
B
0.0  6
1.0  4
```

But if the ALL dtypes aren't numeric:
```python
import pandas as pd

data = {
    "A": ["0.0", "1.0", "2.0", "3.0", "4.0"],
    "B": ["0.0", "1.0", "0.0", "1.0", "0.0"],
    "C": ["foo1", "foo2", "foo3", "foo4", "foo5"],
    # "D": [0, 1, 2, 3, 4],
}
pandas_df = pd.DataFrame(data)
print(f"pandas dtypes:\n {pandas_df.dtypes}")

pandas_res = pandas_df.groupby("B").sum()

print(pandas_res)
```

Nothing to be dropped:
```bash
pandas dtypes:
 A    object
B    object
C    object
dtype: object
             A             C
B
0.0  0.02.04.0  foo1foo3foo5
1.0     1.03.0      foo2foo4
```